pkg/engine/imageVerify.go

```diff
@@ -153,15 +153,35 @@ func (iv *imageVerifier) verify(imageVerify *v1.ImageVerification, images map[st
 				continue
 			}
 
+			if imageVerify.MutateDigest == nil {
+				mutate := true
+				imageVerify.MutateDigest = &mutate
+			}
+
 			var ruleResp *response.RuleResponse
 			if len(imageVerify.Attestations) == 0 {
 				var digest string
 				ruleResp, digest = iv.verifySignature(imageVerify, imageInfo)
-				if ruleResp.Status == response.RuleStatusPass {
-					iv.patchDigest(path, imageInfo, digest, ruleResp)
+				if imageInfo.Digest == "" && *imageVerify.MutateDigest && ruleResp.Status == response.RuleStatusPass {
+					err := iv.patchDigest(path, imageInfo, digest, ruleResp)
+					if err != nil {
+						ruleResp.Message = err.Error()
+						ruleResp.Status = response.RuleStatusFail
+					}
 				}
 			} else {
 				ruleResp = iv.attestImage(imageVerify, imageInfo)
+				if imageInfo.Digest == "" && *imageVerify.MutateDigest {
+					digest, err := fetchImageDigest(imageInfo.String())
+					if err != nil {
+						iv.logger.V(4).Info("fetching image data from registry", "error", err)
+					}
+					err = iv.patchDigest(path, imageInfo, digest, ruleResp)
+					if err != nil {
+						ruleResp.Message = err.Error()
```

Comment:
Please enrich the error with e.g. fmt.Sprintf("failed to patch image with digest: %s", err.Error())